docs/tasks/development-testing.md

```diff
@@ -0,0 +1,37 @@
+# Development Testing
+
+In order to help development for the proxy, there are a few tools in place for
+quick testing.
+
+# Creating a Cluster
+
+Use `make dev_cluster_create` to spin up a kind cluster locally. This will also
+build the proxy and other tooling from source, build their images, and load them
+onto each node.
+
+# Deploying the Proxy
+
+Use `make dev_cluster_deploy` to build the proxy and other tooling from source,
```

Comment:
Perhaps it's worth moving the `make` commands into a expanded (three backtick) block? I almost missed this one when scanning and couldn't work out why `make dev_cluster_create` didn't output a token 🙈

Reply:
Also, I received the following after running `make dev_cluster_create && make dev_cluster_deploy`:

```
fork/exec /Users/james/go/src/github.com/jetstack/kube-oidc-proxy/bin/kubectl: no such file or directory
```

Reply:
```
$ make bin/kubectl
make: *** No rule to make target `bin/kubectl'.  Stop.
```

I think this is because of `BINDIR    ?= $(CURDIR)/bin`, which means that the target name is the fully expanded path (i.e. `/Users/james/go/src/github.com/jetstack/kube-oidc-proxy/bin`)

Reply:
Yep:

```
$ make /Users/james/go/src/github.com/jetstack/kube-oidc-proxy/bin/kubectl
mkdir -p /Users/james/go/src/github.com/jetstack/kube-oidc-proxy/bin
curl --fail -sL -o /Users/james/go/src/github.com/jetstack/kube-oidc-proxy/bin/.kubectl https://storage.googleapis.com/kubernetes-release/release/v1.15.0/bin/darwin/amd64/kubectl
echo "63f1ace419edffa1f5ebb64a6c63597afd48f8d94a61d4fb44e820139adbbe54  /Users/james/go/src/github.com/jetstack/kube-oidc-proxy/bin/.kubectl" | shasum -a 256 -c
/Users/james/go/src/github.com/jetstack/kube-oidc-proxy/bin/.kubectl: OK
chmod +x /Users/james/go/src/github.com/jetstack/kube-oidc-proxy/bin/.kubectl
mv /Users/james/go/src/github.com/jetstack/kube-oidc-proxy/bin/.kubectl /Users/james/go/src/github.com/jetstack/kube-oidc-proxy/bin/kubectl
```

Reply:
```
NewReplicaSet:   oidc-issuer-e2e-5cd6fc49d8 (1/1 replicas created)
Events:
  Type    Reason             Age   From                   Message
  ----    ------             ----  ----                   -------
  Normal  ScalingReplicaSet  21s   deployment-controller  Scaled up replica set oidc-issuer-e2e-5cd6fc49d8 to 1
timed out waiting for the condition
exit status 1
make: *** [dev_cluster_deploy] Error 1
(kind-kube-oidc-proxy-e2e:default)➜  kube-oidc-proxy git:(dev-cluster-deploy-improvments) kc get po -A
NAMESPACE                   NAME                                                        READY   STATUS             RESTARTS   AGE
kube-oidc-proxy-e2e-bhqf4   oidc-issuer-e2e-684979fb5d-6wxp2                            0/1     CrashLoopBackOff   5          4m13s
kube-oidc-proxy-e2e-kppmc   oidc-issuer-e2e-5cd6fc49d8-svd6v                            0/1     Error              2          27s
kube-system                 coredns-6955765f44-prhzr                                    1/1     Running            0          7m32s
kube-system                 coredns-6955765f44-xfvvf                                    1/1     Running            0          7m32s
kube-system                 etcd-kube-oidc-proxy-e2e-control-plane                      1/1     Running            0          7m45s
kube-system                 kindnet-q675q                                               1/1     Running            0          7m16s
kube-system                 kindnet-v9szq                                               1/1     Running            0          7m31s
kube-system                 kube-apiserver-kube-oidc-proxy-e2e-control-plane            1/1     Running            0          7m45s
kube-system                 kube-controller-manager-kube-oidc-proxy-e2e-control-plane   1/1     Running            0          7m45s
kube-system                 kube-proxy-nm4q8                                            1/1     Running            0          7m31s
kube-system                 kube-proxy-zvsfw                                            1/1     Running            0          7m16s
kube-system                 kube-scheduler-kube-oidc-proxy-e2e-control-plane            1/1     Running            0          7m45s
local-path-storage          local-path-provisioner-7745554f7f-zxpjb                     1/1     Running            0          7m32s
(kind-kube-oidc-proxy-e2e:default)➜  kube-oidc-proxy git:(dev-cluster-deploy-improvments) kc logs -n kube-oidc-proxy-e2e-bhqf4 oidc-issuer-e2e-684979fb5d-6wxp2
standard_init_linux.go:211: exec user process caused "exec format error"
```

looks like you need to set GOOS/GOARCH (and possibly CGO_ENABLED) for binaries that are destined to be built into images :) (host vs target platform!)